src/services/goToDefinition.ts

```diff
@@ -26,6 +26,12 @@ namespace ts.GoToDefinition {
             return label ? [createDefinitionInfoFromName(typeChecker, label, ScriptElementKind.label, node.text, /*containerName*/ undefined!)] : undefined; // TODO: GH#18217
         }
 
+        if (node.kind === SyntaxKind.ReturnKeyword) {
+            const functionDeclaration = findAncestor(node.parent, n =>
+                isClassStaticBlockDeclaration(n) ? "quit" : isFunctionLikeDeclaration(n)) as FunctionLikeDeclaration;

```

Comment:
There is technically a `getContainingFunctionOrClassStaticBlock` - but I think this works too.

Reply:
Sorry @DanielRosenwasser, I thought this was good to go so merged.

Reply:
Nah, this is fine I think.

Reply:
`getContainingFunctionOrClassStaticBlock` uses `isFunctionLikeOrClassStaticBlockDeclaration -> isFunctionLikeKind`, which allows to use the following node types
```ts
case SyntaxKind.MethodSignature:
case SyntaxKind.CallSignature:
case SyntaxKind.JSDocSignature:
case SyntaxKind.ConstructSignature:
case SyntaxKind.IndexSignature:
case SyntaxKind.FunctionType:
case SyntaxKind.JSDocFunctionType:
case SyntaxKind.ConstructorType:
```

I think in this case we don't need all of them, or am I wrong?

Reply:
Okay :)